flax/core/frozen_dict.py

```diff
@@ -220,8 +220,13 @@ def _frozen_dict_state_dict(xs):
 
 
 def _restore_frozen_dict(xs, states):
+  diff = set(map(str, xs.keys())).difference(states.keys())
```

Comment:
we might need to be careful about being too strict with dict restores in cases of partial restorations